src/auth/src/credentials/idtoken/impersonated.rs

```diff
@@ -138,19 +138,19 @@ impl Builder {
     /// Target principal is the email of the service account to impersonate.
     ///
     /// # Example
-    /// ```
+    /// ```no_run
```

Comment:
FWIW, `no_run` and putting the example inside the function is overkill.  The code will "run" but the function is not called, it is just defined.

Reply:
And if the code does not need `async` there is no need to put this into a function.